### ISSUE TITLE
aws_network_acl_rule: Transient error on creation

### ISSUE THREAD

**Issue by vancluever** [hashicorp/terraform#4721]
Guys,

Been getting errors like these in my VPC modules:

```
Error applying plan:

1 error(s) occurred:

* aws_network_acl_rule.network_acl_rule.2: Expected to find one Network ACL, got: []*ec2.NetworkAcl(nil)
```

Re-running the `apply` succeeds, without any extra resources created. I'm guessing maybe we are not waiting for resource creation properly?

**Comment by jen20** [comment]
Hi @vancluever, I've put forward a potential fix in #4734 for this. Thanks for reporting!

**Comment by vancluever** [comment]
Thanks @jen20!
